solutions/scaling_microfrontend_aws_eks/index.asciidoc

```diff
@@ -0,0 +1,161 @@
+
+There are different approaches to micro-frontend. They usually try to resolve the challenge of how to put together your independently developed elements and present it as a single application to the end user. One approach is to create reusable custom elements by following the https://developer.mozilla.org/en-US/docs/Web/Web_Components[Web Components] standard. In this appraoch you have different teams develop the custom element they are responsible for and put them together in an encapsulating application which is deployed as a single unit, for example using a docker container. You can follow https://github.com/devonfw/devon4ts/wiki/guide-angular-elements[this guide] to developing custom elements in Angular. One challenge with this approach is that when the demand for one of your micro-frontend app increases, you have to increase the number of containers serving your entire application. For example, considering the e-commerce application we mentioned earlier, the products page might get more hits than the checkout page. But even then you have to scale out the containers serving the entire application. This defeats the purpose of developing each element of your application independently.
+
+Another approach, which resolves the above challenge, is to deploy each micro-frontend app in its own container and serve them along different routes of your domain using Ingress. Given the above scenario, when the demand for products page gets higher, you can scale out the containers serving only the products page, rather than scaling out your entire application. 
+
+image::mfe-with-eks.svg[Micro front-end using EKS, width=700]
+
+For communication between the various micro apps, you can use the publish-subscribe model, query parameters or both in conjunction. This document will focus on this approach using Amazon EKS.
+
+
+=== AWS Elastic Kubernetes Service
+
+Kubernetes is a popular container orchestration tool, which helps in managing numerous containers working in parallel. One of their most useful benefit is scaling out your containers when demand is high. 
+
+AWS EKS is a managed Kubernetes service provided by Amazon. With EKS, you don’t need to install, operate, or maintain the Kubernetes control plane or the worker nodes. EKS provides high availability for both worker and master nodes. The control plane instance run across several Availabiliity Zones. EKS detects and replaces unhealthy nodes automatically and also  provides scalability and security to applications.
```

Comment:
I would just say that we're using a managed Kubernetes in AWS (AWS EKS)
Put links to the product page of Kubernetes and EKS inside. I would not describe them. (Most of the visitors will know it)